addons/account/models/account_journal_dashboard.py

```diff
@@ -694,6 +698,34 @@ def _get_journal_dashboard_bank_running_balance(self):
             )
         return result
 
+    def _get_outstanding_payments_balances(self):
+        self.env.cr.execute("""
+            SELECT move.journal_id AS journal_id,
+                   move.company_id AS company_id,
+                   move.currency_id AS currency,
+                   SUM(CASE
+                       WHEN payment.payment_type = 'outbound' THEN -payment.amount
+                       ELSE payment.amount
+                   END) AS amount_total,
+                   SUM(amount_company_currency_signed) AS amount_total_company
+              FROM account_payment payment
+              JOIN account_move move ON move.payment_id = payment.id
+              JOIN account_journal journal ON move.journal_id = journal.id
+             WHERE payment.is_matched IS TRUE
+               AND move.state = 'posted'
+               AND move.journal_id = ANY(%s)
+               AND move.company_id = ANY(%s)
+               AND payment.outstanding_account_id = journal.default_account_id
```

Comment:
This doesn't have tight bounds, it will indeed be a big performance hit since
* all payments will eventually be matched
* all moves will eventually be posted
* the fact that the account of the payment is the same as the one on the journal is a configuration, it will always be true as well

In the end, all the payments in the journal will be read, which keeps growing without any bounds.

Reply:
I understand the performance issue here, but I don't see which kind of bounds I could use. The configuration `payment.outstanding_account_id = journal.default_account_id` is the one I'm looking for, which results the payment to be matched as soon as it is posted, I don't really see which other conditions would match my case

Reply:
@william-andre I'm not sure we have many other options here...
unless we do some type of async widget to compute the GL balance

Reply:
maybe we can index `account_payment.outstanding_account_id`

Reply:
Indexing doesn't stop the data from growing every day, and the need to read it in the query
If the condition on the account matches once, I expect it to match often.

Reply:
Tested thanks to @william-andre, the index results in a very tiny impact, execution time of approximately 0.8ms for the extra query on a large db (1 second without the index)